drivers/spi/spi_emul.c

```diff
@@ -84,6 +84,29 @@ static int spi_emul_io(const struct device *dev,
 	return api->io(emul, config, tx_bufs, rx_bufs);
 }
 
+#ifdef CONFIG_SPI_ASYNC
+static int spi_emul_io_async(const struct device *dev,
+		       const struct spi_config *config,
+		       const struct spi_buf_set *tx_bufs,
+		       const struct spi_buf_set *rx_bufs,
+				       struct k_poll_signal *async)
+{
+	struct spi_emul *emul;
+	const struct spi_emul_api *api;
+
+	emul = spi_emul_find(dev, config->slave);
+	if (!emul) {
+		return -EIO;
+	}
+
+	api = emul->api;
+	__ASSERT_NO_MSG(emul->api);
+	__ASSERT_NO_MSG(emul->api->io_async);
```

Comment:
nit:
1. Since `emul->api` is already stored in `api`, just check that variable.
2. Per coding guidelines can we have explicit conditionals? `emul->api != NULL` etc